net-p2p/cardano-db-sync/files/cardano_db_sync.in

```diff
@@ -38,59 +38,127 @@
+jail_state="/${_net}-state"
+jail_args="name=cardano_db_sync_jail exec.jail_user=cardano exec.system_jail_user host=inherit"
+
+jail_root="${_home}/jail"
+jail_copy_resolv_conf=yes
+jail_copy_services=yes
+jail_copy_programs="$command /usr/sbin/nologin /bin/sh /usr/local/bin/psql"
+jail_mount_devfs=no
+jail_ip_inherit=yes
+jail_prepare_inside_cmds="mkdir ./tmp ;\
+                          cp /etc/pwd.db ./etc ;\
+                          cp /etc/spwd.db ./etc ;\
+                          cp /etc/master.passwd ./etc ;\
+                          cp /etc/passwd ./etc ;\
+                          chmod +s ./bin/cardano-db-sync"
+jail_nullfs_mounts="`dirname ${_config}` ./${_net}-configs rw \
+                    ${_schema} ./schema ro \
+                    ${_state} ./${_net}-state rw \
+                    ${_home}/logs ./logs rw"
+
+
+export PGPASSFILE=${jail_pgpass}
 pidfile="/var/run/cardano-db-sync.pid"
-logfile="/var/log/cardano-db-sync.log"
-flags=" --schema-dir ${cardano_db_sync_schema} \
-        --state-dir ${cardano_db_sync_state} \
-        --socket-path ${cardano_db_sync_cnode_socket} \
-        --config ${cardano_db_sync_config} \
+#logfile="/var/log/cardano-db-sync.log"
+flags=" --schema-dir ${jail_schema} \
+        --state-dir ${jail_state} \
+        --socket-path ${jail_socket} \
+        --config ${jail_config} \
         ${cardano_db_sync_flags}"
 
+. %%LOCALBASE%%/share/rc-subr-jail/rc.subr.jail
+
+# realpath2 path
+# Returns an absolute path to ${path} if it exists, otherwise
+# treats ${path} as absolute
+dirname_realpath()
+{
+    local _path _dirname _realpath
+    _path=$1
+    _dirname=$(dirname ${_path})
+
+    _realpath=$(/bin/sh -c "cd $_dirname && readlink ${_path}" 2> /dev/null)
+    if [ $? = "0" ]
+    then
+        _dirname=$(dirname ${_realpath})
+        if [ $_dirname == "." ]
+        then
+            echo $(dirname ${_path})
+        else
+            echo $(/bin/sh -c "cd ${_dirname} && pwd" 2> /dev/null)
+        fi
+        return 0
+    fi
+    echo $(dirname ${_path})
+}
+
 sanity_check()
 {
-    if [ ! -f ${cardano_db_sync_config} ]
+    if [ ! -f ${_config} ]
     then
-        echo "Invalid value for cardano_db_sync_config: missing file ${cardano_db_sync_config}"
+        echo "Invalid value for cardano_db_sync_config: missing file ${_config}"
         echo "You might want to run service cardano_db_sync onefetch"
         exit 1
     fi
-    if [ ! -f ${cardano_db_sync_pgpass} ]
+    if [ ! -f ${_pgpass} ]
     then
-        echo "Invalid value for cardano_db_sync_pgpass: missing file ${cardano_db_sync_pgpass}"
+        echo "Invalid value for cardano_db_sync_pgpass: missing file ${_pgpass}"
         echo "Did you setup postgresql database access?"
         exit 1
     fi
-    if [ ! -d `dirname ${cardano_db_sync_cnode_socket}` ]
+    if [ ! \( -L ${_socket} -o -S ${_socket} \) ]
+    then
+        echo "Invalid value for cardano_db_sync_cnode_socket: missing file ${_socket}"
```

Comment:
`"Invalid value for cardano_db_sync_cnode_socket: ${_socket} is not a socket or a symlink"`